### ISSUE TITLE
Backend configuration does not properly support boolean values

### ISSUE THREAD

**Issue by BlacCello** [craftypath/gotf#134]
Hi,
thanks for sharing the project, the idea is quite promising.
I used gotf to try it in one of our projects.
We use azurerm as backend and I applied to following configuration in gotf.yaml for the state configuration:
```yaml
backendConfigs:
  key: "{{ .Vars.state_key }}"
  storage_account_name: "xxx"
  resource_group_name: "xxx"
  container_name: "xxx"
  snapshot: true
```
On every gotf command, it complains that I have to reconfigure the backend.

```
Error: configured backend does not match current environment

snapshot: got=true, want=true

Run terraform init -reconfigure!
```
It does not matter if I set snapshot to `true` or `'true'`
There is probably a problem with the datatypes of string vs. bool.

```
gotf --version
v0.17.0 (commit=49e4788, date=2022-02-09T17:18:13Z)
```

**Comment by unguiculus** [comment]
Thanks for reporting this. I'm sorry this went unnoticed for quite some time. The internal datatype is a string. I'm gonna look into this.

**Comment by unguiculus** [comment]
As a temporary workaround, you should be able to use the `ARM_SNAPSHOT` environment variable to configure snapshotting.

```yaml
envs:
  ARM_SNAPSHOT: "true"
 
backendConfigs:
  key: "{{ .Vars.state_key }}"
  storage_account_name: "xxx"
  resource_group_name: "xxx"
  container_name: "xxx"
```

Alternatively, you could disable the backend check using the `-s, --skip-backend-check` flag. I wouldn't really recommend that, though.

**Comment by unguiculus** [comment]
@BlacCello Fixed in release [v0.18.0](https://github.com/craftypath/gotf/releases/tag/v0.18.0). Enjoy!
